server/src/main/java/org/apache/cassandra/sidecar/cluster/CQLSessionProviderImpl.java

```diff
@@ -203,4 +248,85 @@ public void close()
             }
         }
     }
+
+    private SslContext createSslContext(SslConfiguration sslConfiguration)
+    {
+        if (sslConfiguration == null || !sslConfiguration.enabled())
+        {
+            return null;
+        }
+
+        SslContextBuilder sslContextBuilder;
+        try
+        {
+            sslContextBuilder = SslContextBuilder.forClient()
+                                                 .protocols(sslConfiguration.secureTransportProtocols());
+
+            if (sslConfiguration.isKeystoreConfigured())
+            {
+                KeyStore keyStore = createKeystore(sslConfiguration.keystore());
+                KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                kmf.init(keyStore, sslConfiguration.keystore().password().toCharArray());
+                sslContextBuilder.keyManager(kmf);
+            }
+
+            if (sslConfiguration.isTrustStoreConfigured())
+            {
+                KeyStore truststore = createKeystore(sslConfiguration.truststore());
+                TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                tmf.init(truststore);
+                sslContextBuilder.trustManager(tmf);
+            }
+            return sslContextBuilder.build();
+        }
+        catch (Exception e)
+        {
+            throw new ConfigurationException("Error creating SsLContext for Cassandra connections", e);
+        }
+    }
+
+    private KeyStore createKeystore(KeyStoreConfiguration config)
+    throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException
+    {
+        KeyStore keystore = KeyStore.getInstance(config.type());
+        try (FileInputStream inputStream = new FileInputStream(config.path()))
+        {
+            keystore.load(inputStream, config.password().toCharArray());
+        }
+        return keystore;
+    }
+
+    /**
+     * {@link MtlsAuthProvider} is a custom AuthProvider. It is required when driver needs to connect to Cassandra with
+     * mutual TLS.
+     */
+    private static class MtlsAuthProvider implements AuthProvider
+    {
+        @Override
+        public Authenticator newAuthenticator(InetSocketAddress inetSocketAddress, String s) throws AuthenticationException
+        {
+            return new MutualTLSAuthenticator();
+        }
+
+        private static class MutualTLSAuthenticator implements Authenticator
+        {
+            @Override
+            public byte[] initialResponse()
+            {
+                return new byte[]{ 0, 0 };
```

Comment:
Just curious. Why is the initial response a `0, 0` here?

Reply:
We figured it wasn't necessary, even a empty array would work. But removed this `MtlsAuthProvider`